### PR TITLE
ARROW-5952: [Python] fix conversion of chunked dictionary array with 0 chunks

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
https://issues.apache.org/jira/browse/ARROW-5952

### CODE REVIEW

**Comment by kszucs** [comment]
@ursabot build

File: cpp/src/arrow/python/arrow_to_pandas.cc [hunk 1]
```diff
@@ -1063,6 +1063,10 @@ class CategoricalBlock : public PandasBlock {
     using T = typename TRAITS::T;
     constexpr int npy_type = TRAITS::npy_type;
 
+    if (data->num_chunks() <= 0) {
```
Comment (wesm) [inline cpp/src/arrow/python/arrow_to_pandas.cc]: nit: num_chunks() cannot be negative (even though it's a signed integer), so checking `== 0` is better

**Review by wesm — COMMENTED** [review]
Thanks for looking into this. It's a weird degenerate case, I'm not sure if there's a better way to handle it

**Comment by jorisvandenbossche** [comment]
Thanks for the review and fixes.

> It's a weird degenerate case, I'm not sure if there's a better way to handle it

Yes, I agree. I needed to put special cases in several places, which didn't feel that good, but I also didn't see a better way.

**Pull request closed without merging**